tests/test_mda.py

```diff
@@ -98,6 +98,116 @@ def cb(img, event):
         assert not core.mda._canceled
 
 
+def test_autofocus(core: CMMCorePlus, qtbot: QtBot, mock_fullfocus):
+    # mock_autofocus sets z=100
+    mda = MDASequence(
+        stage_positions=[{"z": 50}],
+        autofocus_plan={
+            "autofocus_device_name": "Z",
+            "autofocus_motor_offset": 50,
+            "axes": ("p",),
+        },
+    )
+    with qtbot.waitSignals(
+        [
+            core.mda.events.frameReady,
+            core.mda.events.sequenceFinished,
+        ]
+    ):
+        core.run_mda(mda)
+
+    assert core.getPosition() == 100
```

Comment:
can you add a comment here to explain what this is testing?  My only question upon looking at it is whether this pretty much just ends up testing that mock_fullfocus ran?  or whether it's testing something deeper about the autofocus process?